### PR TITLE
Implement send and receive function #32

### PR DESCRIPTION
**Pull request by Harshi7016**
Created with new branch after discussion with @Kratika0907 , have tried to not keep lots of changes in the main index page as per discussion and UI changes and warnings/error messages are added.
Still working on submit handler

### CODE REVIEW

**Comment by Harshi7016** [comment]
@Kratika0907 @ankushdharkar , I have merged Kratika's code and have created this PR....The file changes includes both PR

File: components/coins-status/useCoinStatus.js [hunk 1]
```diff
@@ -1,28 +1,25 @@
...
+  const [start, setStart] = useState(startPoint);
+  const [end, setEnd] = useState(endPoint);
+  let startTimestamp = useRef(null);
+  const step = (timestamp) => {
+    if (!startTimestamp.current) startTimestamp.current = timestamp;
+    const progress = Math.min(
+      (timestamp - startTimestamp.current) / duration,
+      1
+    );
+    setStart(Math.floor(progress * (end - startPoint) + startPoint));
```
Comment (sumitd94) [inline components/coins-status/useCoinStatus.js:13]: NIT : The operation which you are doing inside of `setStart` method, may be do that outside of it and store the value in a const and then pass it to the `setStart` method
This makes the code more readable.

File: components/operationModal/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import TransactionOperation from '@components/transaction-operation';
+import React, { useEffect, useState } from 'react';
+import receivers from '../../mock/receivers';
+const OperationModal = (props) => {
+  const { transactionType, personData } = props;
```
Comment (sumitd94) [inline components/operationModal/index.js:5]: NIT : the destructuring can be done right on line no 4
```javascript
const OperationalModal = ({ transactionType, personData }) => {
```

File: components/operationModal/index.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  const [modal, showModal] = useState(false);
+  useEffect(() => {
+    console.log(transactionType);
```
Comment (sumitd94) [inline components/operationModal/index.js:8]: Please make sure to remove console statements

File: components/transaction-chart/index.js [hunk 4]
```diff
@@ -40,13 +40,12 @@ function getDataset(transactionData) {
...
-    setChartData(getDataset(props.transactionChartData));
-  };
-
   useEffect(() => {
+    const chart = () => {
+      setChartData(getDataset(props.transactionChartData));
+    };
     chart();
-  }, [chart]);
+  }, [props.transactionChartData]);
```
Comment (sumitd94) [inline components/transaction-chart/index.js:48]: NIT: Best practice will be to use destructuring

File: components/coins-status/useCoinStatus.js (hunk 1 above)
Comment (Harshi7016) [reply components/coins-status/useCoinStatus.js:13]: Thanks @sumitd94 for the review, I will make these changes 👍

File: components/operationModal/index.js (hunk 2 above)
Comment (Harshi7016) [reply components/operationModal/index.js:5]: I was bit confused whether that is a good practise or writing it down, I will follow your approach from now onwards

File: components/operationModal/index.js (hunk 3 above)
Comment (Harshi7016) [reply components/operationModal/index.js:8]: Good catch sir

File: components/transaction-chart/index.js (hunk 4 above)
Comment (Harshi7016) [reply components/transaction-chart/index.js:48]: You are right, I will update that

File: pages/index.js [hunk 5]
```diff
@@ -6,12 +6,15 @@ import transactionChartData from '../mock/transaction-graph-data.json';
 import TransactionList from '../components/transaction-details';
 import coinsData from '../mock/coins.json';
 import CoinsStatus from '../components/coins-status';
-import PersonDetail from '../components/user-info';
 import { Footer } from '../components/footer';
 import styles from '../styles/Home.module.css';
 import TransactionChart from '../components/transaction-chart';
+import OperationModal from '@components/operationModal';
+import { useState } from 'react';
```
Comment (sumitd94) [inline pages/index.js:13]: Please declare this on first line.
If you are importing anything from react, it should be declared right at the top

Comment (Kratika0907) [reply pages/index.js:13]: Yes for code readability imports should be mentioned in this order -> external, internal components & libs , style sheets

Comment (Harshi7016) [reply pages/index.js:13]: Oh, I will keep this in mind from now onwards.... Thanks guys for your time

File: components/operationModal/index.js (hunk 2 above)
Comment (sumitd94) [reply components/operationModal/index.js:5]: The approach you followed makes sense if you have like say 8-9 props coming in ..then for code readability, its better to destructure it in a new line .. but for 3-4 props, its better to destructure it like I mentioned :)

File: pages/index.js (hunk 5 above)
Comment (Harshi7016) [reply pages/index.js]: @sumitd94 changes are added

**Comment by Harshi7016** [comment]
@sumitd94  I have resolved all the requested changes

**Comment by Kratika0907** [comment]
@Harshi7016  Please resolve the conflicts and we can close this. Sorry for the delay.

File: components/transaction-operation-modal/index.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import TransactionOperation from '@components/transaction-operation';
+import React, { useEffect, useState } from 'react';
+import receivers from '../../mock/receivers';
+import styles from '../../styles/Home.module.css';
+const TransactionOperationModal = (props) => {
+  const { personData } = props;
```
Comment (swarajpure) [inline components/transaction-operation-modal/index.js:6]: How about 
```const TransactionOperationModal = ({ personData }) => {``` ?

**Review by Kratika0907 — APPROVED** [review]

**Pull request merged**